### PR TITLE
feat: add integration tests for JSON type support

### PR DESCRIPTION
**Pull request by zoercai**
Integration Tests for the new JSON type, which allows users to read and write to Cloud Spanner databases using the JSON type through the client libraries.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java [hunk 1]
```diff
@@ -225,6 +226,23 @@ public void bindStringNull() {
     assertThat(row.isNull(0)).isTrue();
   }
 
+  @Test
+  public void bindJson() {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java]: Could for the sake of completeness also add a test for a non-empty, non-null JSON value?

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java [hunk 2]
```diff
@@ -427,6 +445,37 @@ public void bindStringArrayNull() {
     assertThat(row.isNull(0)).isTrue();
   }
 
+  @Test
+  public void bindJsonArray() {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+    Struct row =
+        execute(
+            Statement.newBuilder("SELECT @v").bind("v").toJsonArray(asList("{}", "[]", null)),
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java]: nit: Add a non-empty JSON value to the array?

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java [hunk 3]
```diff
@@ -320,6 +323,23 @@ public void writeStringNull() {
     assertThat(row.isNull(0)).isTrue();
   }
 
+  @Test
+  public void writeJson() {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java]: nit: Add a test for non-empty, non-null JSON value.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java [hunk 4]
```diff
@@ -320,6 +323,23 @@ public void writeStringNull() {
...
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getJson(0)).isEqualTo("{}");
+  }
+
+  @Test
+  public void writeJsonNull() {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+    write(baseInsert().set("JsonValue").to(Value.json(null)).build());
+    Struct row = readLastRow("JsonValue");
+    assertThat(row.isNull(0)).isTrue();
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITWriteTest.java]: nit: Also add an assertion that the type of the column is recognized as JSON. Considering the fact that this type partly overlaps with `String`, I think it's good to have an assertion that verifies that we can never accidentally make any changes that would make the distinction between a null `String` and null `JSON` fail.

**Review by olavloite — APPROVED** [review]
This looks generally good to me, with a couple of nits for adding some additional values to the test cases.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java [hunk 5]
```diff
@@ -225,6 +226,23 @@ public void bindStringNull() {
...
+  public void bindJson() {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+    Struct row =
+        execute(Statement.newBuilder("SELECT @v").bind("v").to(Value.json("{}")), Type.json());
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getJson(0)).isEqualTo("{}");
+  }
+
+  @Test
+  public void bindJsonNull() {
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java]: nit: per convention we usually name the tests as `testBindJsonNull`

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java [hunk 6]
```diff
@@ -427,6 +445,37 @@ public void bindStringArrayNull() {
...
+
+  @Test
+  public void bindJsonArrayEmpty() {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+    Struct row =
+        execute(
+            Statement.newBuilder("SELECT @v").bind("v").toJsonArray(Collections.emptyList()),
+            Type.array(Type.json()));
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getJsonList(0)).containsExactly();
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java]: nit: this assertion reads a bit funny, could you do an `assertThat(...).isEqualTo(Collections.emptyList())` instead?

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java (hunk 5 above)
Comment (zoercai) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java]: this would make the JSON test names different from the other test names in this test class, is that okay?

**Comment by zoercai** [comment]
Closing as I've added the tests into the implementation PR
